### PR TITLE
Extend select-to-save flashcards to lesson exercises and conversation transcripts

### LINKED ISSUE #314: Extend select-to-save flashcards to lesson exercise texts and live conversation transcript
#### Is your feature request related to a problem?

FreeLingo already lets users select a word to save it as a vocabulary flashcard in text chat, lesson explanations, and reading/listening exercise prompts (as summarized in #292). However, two of the surfaces where learners most often meet unknown words do not support it:

- **Lesson exercises**: on the lesson screen the explanation text is selectable, but the exercise question text below it is not (`frontend/src/app/(app)/lesson/[id]/page.tsx` — the explanation block is wired with `word-selectable` + `WordTooltip`, while `exercise.question` is rendered as plain text). Selecting a word works in one half of the screen and silently does nothing in the other, and exercises are precisely where a learner proves a word is unknown.
- **Live voice conversation**: transcript entries in `ConversationMode` have no word-save at all, so vocabulary introduced by the tutor during a conversation cannot be captured. #292 already listed extending word-save to voice transcripts as a candidate for future development; this issue picks that part up.

#### Proposed Solution

Reuse the existing selection flow (`useWordSave` + `WordTooltip` + `POST /api/flashcards/from-word`) on both surfaces — the same interaction users already know from reading/listening/chat, working with mouse and touch via `onPointerUp`.

#### Key Functionality

- **Lesson exercises**: make the exercise question/prompt text word-selectable. Answer options (multiple choice) and user input stay non-selectable, consistent with the existing design for reading and listening exercises.
- **Voice conversation**: make transcript entries word-selectable (at least tutor turns). Selection saves the word with the surrounding sentence as context, like the other surfaces.
- **Deduplicate `POST /flashcards/from-word`**: unlike `/flashcards/bulk`, this endpoint currently creates a duplicate card when the same normalized word is saved again. Since this change funnels more saves through it, it should recognize the word and show the "already saved" state instead of inserting a duplicate. Happy to split this into a separate bug report if preferred.

#### Alternatives considered

A right-click "Add to flashcards" context menu was considered and rejected: it has no touch equivalent on mobile, conflicts with the native browser context menu, and would introduce a second interaction pattern for an action the app already exposes via selection.

#### User Value

Words encountered mid-exercise or mid-conversation — exactly the ones a learner demonstrably does not know — can be captured for spaced repetition on the spot, using the interaction that already exists elsewhere in the app, instead of relying on topic-based set generation, which may produce already-known words.

### PR DESCRIPTION
**Pull request by arqo123**
Closes #314

## What

Extends the existing select-to-save flashcard flow to the two surfaces where learners most often meet unknown words, and adds best-effort deduplication to the underlying endpoint:

- **Lesson exercise questions** (`lesson/[id]`): the exercise question text is now word-selectable with the same `WordTooltip` pattern used by the explanation block. Answer options, hints, and user input stay non-selectable, consistent with the reading/listening design. The tooltip is dismissed automatically when navigating to the next exercise.
- **Voice conversation transcript** (`ConversationMode`): tutor turns are now word-selectable (user turns and the in-progress streaming bubble are not, matching the chat page where only assistant messages are selectable).
- **`POST /api/flashcards/from-word` dedupe**: the endpoint now checks the input word before calling the LLM (skipping the lookup entirely on a hit) and re-checks the canonicalized result word after the lookup. Each check is a single targeted query (`lower(trim(word))` on the user's active plan, oldest card wins) that returns the matching row itself, so nothing is scanned or materialized. An existing `from_text` card comes back with `already_saved=true`; a matching generated or imported card (`source=None`) is promoted to `from_text` in place and returned with `already_saved=false`, so the word appears in My Vocabulary without creating a duplicate in the review deck.
- **New "already saved" tooltip state**: when the backend reports `already_saved`, the tooltip shows a neutral "already saved" confirmation instead of the green "saved" state. The new label is localized in all ten languages.
- **Tooltip lifecycle**: `useWordSave` ignores responses and auto-dismiss timers that belong to a previous selection (selecting B while A is still saving never marks B as saved). `ConversationMode` dismisses the tooltip whenever the transcript mutates, on session start, and on session finalize. The lesson page dismisses it only when the active exercise changes, not when answers, hints, or regenerations replace items in the exercises array.

## Screenshots

Selecting a word in a lesson exercise question:

Lesson exercise tooltip

Saving a word that is already in the vocabulary (deduplicated, no LLM call):

Already saved state

Selecting a word in a tutor turn of the voice-conversation transcript:

Conversation transcript tooltip

## Tests

- Backend: `pytest` — 1026 passed. Seven new/updated tests cover the dedupe paths: existing input word skips the LLM and returns `already_saved`, lemma collision after lookup does not insert, fresh words insert normally, a matching word under a *different* study plan does not suppress the save (pins the per-plan scoping), a `source=None` collision is promoted to `from_text` both before and after the LLM call, and pre-existing duplicates deterministically resolve to the oldest card.
- Frontend: `npm run lint`, `tsc --noEmit`, `vitest run` — 498 tests passed across 52 files, including tests for the `WordTooltip` states (`exists` rendering, `already_saved` response handling, auto-dismiss, stale response and timer isolation between selections), `TranscriptBubble` selectable wiring, a `ConversationMode` integration test (tooltip dismissed on new turn, stop, and restart), and a lesson page integration test (tooltip survives an `exercises` mutation, is dismissed on navigation).

## Docs

`specs/api-endpoints.instructions.md`, `specs/architecture-frontend.instructions.md`, and the AGENTS.md architecture summary line were updated to match. `CHANGELOG.md` and `specs/version.md` were intentionally left untouched — release notes and version bumps are maintainer territory and conflict across parallel PRs.

## Notes

Deduplication is best-effort, not strict idempotency: two concurrent first-time saves of the same word within milliseconds of each other can still both insert (the post-lookup re-check narrows the window the slow LLM call opens, but only a normalized per-user/plan uniqueness constraint would fully close it). That would require a migration, so it is left out of scope here — happy to follow up if you'd like one.

### CODE REVIEW

**Review by artcc — CHANGES_REQUESTED** [review]
La dirección general es correcta: cubre las dos superficies solicitadas, mantiene opciones/inputs/turnos del usuario y streaming fuera de la selección, conserva el aislamiento por usuario y plan, contempla la canonicalización posterior al LLM y añade un contrato de respuesta compatible. Los jobs de backend y frontend también pasan en el SHA revisado.

Solicito cambios por los comentarios inline antes de fusionar:

- Corregir la colisión con tarjetas que no son `from_text`, para que “Already saved” no deje la palabra fuera de “My Vocabulary”.
- Evitar los dos escaneos completos de todas las tarjetas del plan por guardado.
- Cerrar/invalidar correctamente el tooltip cuando cambia o se reinicia el transcript.
- Proteger `useWordSave` frente a respuestas y temporizadores de selecciones anteriores.
- Limitar el cierre del tooltip de lección a la navegación real entre ejercicios.
- Endurecer la recuperación de coincidencias frente a borrado concurrente y selección no determinista de duplicados existentes.

La carrera entre dos inserciones simultáneas ya está reconocida en la descripción. Puede quedar como follow-up si se presenta la deduplicación como best-effort y no como idempotencia estricta; una garantía completa requeriría bloqueo o unicidad normalizada por usuario y plan.

Faltan pruebas de integración para `ConversationMode` y la página de lección, además de casos de colisión `source=None`, selección A→B con A pendiente y carreras. Los tests actuales cubren bien los caminos básicos y componentes aislados, pero no estos ciclos de vida.

El propietario del repositorio gestionará por separado changelog, versión, specs y actualización de métricas de tests, por lo que no los incluyo como cambios solicitados al autor.

File: backend/app/routers/flashcards.py [hunk 1]
```diff
@@ -38,6 +39,24 @@ def _normalize_flashcard_word(value: str) -> str:
     return re.sub(r"\s+", " ", value.strip().lower())
 
 
+async def _index_existing_words_by_id(
+    db: AsyncSession, user_id: int, study_plan_id: int
+) -> dict[str, int]:
+    """Map normalized word -> flashcard id for a user's plan (first wins)."""
+    existing = await db.execute(
+        select(Flashcard.id, Flashcard.word).where(
+            Flashcard.user_id == user_id,
```
Comment (artcc) [inline backend/app/routers/flashcards.py:57]: **The deduplication crosses sources with different semantics.** This query includes generated or imported cards with `source = NULL`, while `GET /api/flashcards/vocabulary` only returns cards with `source == "from_text"`. If a word already exists as a generated card, selecting it returns `already_saved=true` even though it remains absent from My Vocabulary, which contradicts the UI flow and copy. When an existing match is explicitly saved, it should either be promoted to `source="from_text"`, or this deduplication should be limited to cards visible in that list. Please add a collision test with `source=None`.

File: frontend/src/components/conversation/ConversationMode.tsx [hunk 2]
```diff
@@ -301,6 +302,14 @@ export default function ConversationMode({
   } = useTransientToast()
   const [quota, setQuota] = useState<QuotaStatus | null>(null)
   const [reviewPromptOpen, setReviewPromptOpen] = useState(false)
+  const {
+    selectedWord,
+    tooltipPos,
+    saveState,
```
Comment (artcc) [inline frontend/src/components/conversation/ConversationMode.tsx:308]: **The tooltip dismissal is not integrated with the conversation lifecycle.** `handleStart()` clears `transcript` without calling `dismissTooltip()`, and auto-scroll moves bubbles while the tooltip keeps fixed viewport coordinates. A user can select a word, end and start another session, and still see or save context from the previous transcript; new turns can also leave the tooltip positioned over unrelated content. Please dismiss it when restarting or finalizing a session and whenever a transcript mutation invalidates the selection. Add a `ConversationMode` integration test, because the isolated `TranscriptBubble` test does not cover this behavior.

File: frontend/src/components/ui/WordTooltip.tsx [hunk 3]
```diff
@@ -132,7 +142,8 @@ export function useWordSave() {
         }),
       })
```
Comment (artcc) [inline frontend/src/components/ui/WordTooltip.tsx:170]: **A stale response can update a newer selection.** If word A is being saved and word B is selected before A completes, the response for A marks the tooltip for B as `saved` or `exists`; the timeout created for A then dismisses B. This partly predates the PR, but this change expands the hook to another dynamic surface and the new state follows the same path. Please associate each request and timeout with the active selection, ignore stale responses, and cancel the previous timeout on selection, dismissal, or unmount. Add a test for selecting B while A is still pending.

File: backend/app/routers/flashcards.py [hunk 4]
```diff
@@ -38,6 +39,24 @@ def _normalize_flashcard_word(value: str) -> str:
...
+            Flashcard.study_plan_id == study_plan_id,
+        )
+    )
+    index: dict[str, int] = {}
```
Comment (artcc) [inline backend/app/routers/flashcards.py]: **This operation materializes the entire plan and runs twice on every miss.** Although only two columns are selected, a new word transfers and normalizes every flashcard before and after the LLM call. The `/bulk` pattern amortizes that cost across a batch; here, with large plans and a 30/minute limit, it introduces O(n) growth per save and avoidable database, CPU, and memory pressure. Please use a targeted query for the normalized form or an indexable normalized field instead of two full scans.

File: frontend/src/app/(app)/lesson/[id]/page.tsx [hunk 5]
```diff
@@ -299,7 +299,8 @@ export default function LessonPage() {
         : ''
     )
     setRegenerateError(null)
-  }, [currentExercise, exercises])
+    dismissTooltip()
```
Comment (artcc) [inline frontend/src/app/(app)/lesson/[id]/page.tsx:308]: **This dismisses the tooltip on every `exercises` change, not only on navigation.** Answering, regenerating, or receiving a native hint or explanation replaces items in the array and triggers this effect even when the active question has not changed. A help request started before selecting a word can also dismiss the tooltip when it resolves. Please separate answer restoration, which depends on `exercises`, from navigation dismissal, which should depend only on `currentExercise` or run directly during the transition.

File: backend/app/routers/flashcards.py [hunk 6]
```diff
@@ -280,6 +299,15 @@ async def create_flashcard_from_word(
     db: AsyncSession = Depends(get_db),
 ):
     plan = await _get_active_plan_or_404(db, current_user.id)
+    existing_by_word = await _index_existing_words_by_id(db, current_user.id, plan.id)
+
+    normalized_input = _normalize_flashcard_word(data.word)
+    if normalized_input in existing_by_word:
+        card = await db.get(Flashcard, existing_by_word[normalized_input])
```
Comment (artcc) [inline backend/app/routers/flashcards.py]: **There is a race between building the index and retrieving the card.** `db.get()` can return `None` if another request deletes the card between those queries, and `FlashcardFromWordResponse.model_validate(None)` would result in a 500. Please retrieve the row in the deduplication query itself or handle `None` by continuing with lookup or insertion. Also, because duplicates may already exist and the query has no ordering, the documented first-wins behavior is currently nondeterministic.

**Review by artcc — CHANGES_REQUESTED** [review]
The overall direction is correct: it covers both requested surfaces, keeps answer options, inputs, user turns, and streaming turns outside selection, preserves user and study-plan isolation, handles post-LLM canonicalization, and adds a backward-compatible response contract. Both backend and frontend checks also pass on the reviewed SHA.

Changes requested before merge:

- Fix collisions with cards that are not `from_text`, so “Already saved” does not leave the word absent from My Vocabulary.
- Avoid two full scans of every card in the plan per save.
- Correctly dismiss or invalidate the tooltip when the transcript changes or restarts.
- Protect `useWordSave` from responses and timers belonging to previous selections.
- Limit lesson tooltip dismissal to actual exercise navigation.
- Harden match retrieval against concurrent deletion and nondeterministic selection among existing duplicates.

The race between two simultaneous insertions is already acknowledged in the PR description. It can remain a follow-up if the behavior is described as best-effort deduplication rather than strict idempotency; a full guarantee requires locking or normalized uniqueness per user and plan.

Integration tests are still needed for `ConversationMode` and the lesson page, along with cases for `source=None` collisions, selecting B while A is pending, and race behavior. The current tests cover the basic paths and isolated components well, but not these lifecycle cases.

The repository owner will handle the changelog, version, specs, and documented test metrics separately, so those are not requested from the contributor.

File: backend/app/routers/flashcards.py (hunk 1 above)
Comment (arqo123) [reply backend/app/routers/flashcards.py:57]: Good catch. I went with promotion rather than narrowing the check: limiting the dedup to `from_text` would leave a generated card and a `from_text` card for the same word side by side in the review deck, which is exactly what `/bulk` avoids.

Now a matching card with `source=None` gets `source="from_text"` set in place and comes back with `already_saved=false`, so the UI shows "Saved" and the word shows up in My Vocabulary. `already_saved=true` is reserved for cards that were already visible there, which keeps the flag consistent with the copy.

Tests: `test_create_flashcard_from_word_generated_card_collision_promotes_source` (pre-LLM path, asserts `/vocabulary` lists the card afterwards and the row count stays at 1) and `test_create_flashcard_from_word_lemma_collision_with_generated_card_promotes` (post-LLM path).

File: frontend/src/components/conversation/ConversationMode.tsx (hunk 2 above)
Comment (arqo123) [reply frontend/src/components/conversation/ConversationMode.tsx:308]: Done. The tooltip is now dismissed in three places:

- in the auto-scroll effect (keyed on `transcript` / `streamingText`), so any new turn or streaming token that moves the bubbles drops it together with the stale coordinates;
- explicitly in `handleStart()` next to the transcript reset;
- in `finalizeSession()`, so every way a session ends (manual stop, server `session_end`, WebSocket error or close) clears it as well.

One detail: `dismissTooltip()` now also runs on every streaming token, so I guarded the `removeAllRanges()` call inside the hook to only fire when a tooltip was actually open. Otherwise the tutor streaming would wipe any text the user is selecting elsewhere on the page.

Added `tests/components/ConversationMode.test.tsx`, which renders the real component with a mocked WebSocket / VAD / AudioContext and drives it through the session: new turn → tooltip gone, stop → tooltip gone, restart → tooltip gone and the previous transcript cleared. All three fail against the previous code.

File: frontend/src/components/ui/WordTooltip.tsx (hunk 3 above)
Comment (arqo123) [reply frontend/src/components/ui/WordTooltip.tsx:170]: Done. `useWordSave` now keeps a selection counter in a ref: every accepted selection and every dismissal bumps it, `handleSaveWord` captures the value it started with and drops the response (success or error) if the counter moved in the meantime. The auto-dismiss timeout lives in a ref too: it is cleared on a new selection, on dismissal and on unmount, and re-checks the counter before firing.

Three new tests in `WordTooltip.test.tsx`: B selected while A is pending (A's late response and timer leave B untouched), dismiss while a save is pending, and a new selection cancelling the previous dismiss timer.

File: backend/app/routers/flashcards.py (hunk 4 above)
Comment (arqo123) [reply backend/app/routers/flashcards.py]: Replaced both scans with one targeted query per check:

```sql
WHERE user_id = ? AND study_plan_id = ? AND lower(trim(word)) = :normalized
ORDER BY id LIMIT 1
```

It rides on the existing `user_id` / `study_plan_id` indexes and returns at most one row, so nothing is materialized on either side of the LLM call. I did not add a normalized column, since that needs a migration and the targeted query already removes the O(n) transfer; happy to do that separately if you would rather have an indexable field.

One caveat: case folding now happens in the database (`lower()`, Unicode-aware on Postgres, same as the `ilike` search in `/vocabulary`), and only trim + lower are reproduced in SQL, not the whitespace collapse from `_normalize_flashcard_word`. The endpoint only ever receives single tokens (the frontend rejects selections containing whitespace), so it makes no practical difference, but it is a small divergence from `/bulk`.

File: frontend/src/app/(app)/lesson/[id]/page.tsx (hunk 5 above)
Comment (arqo123) [reply frontend/src/app/(app)/lesson/[id]/page.tsx:308]: Split it. Answer restoration keeps `[currentExercise, exercises]`, and the dismissal now lives in its own effect keyed on `currentExercise` only, so answering, regenerating, or a hint / explanation resolving no longer closes the tooltip.

Added `tests/app/lesson-word-tooltip.test.tsx`, which renders the real page: selecting a word and then submitting the answer (which replaces the item in `exercises`) keeps the tooltip open, and clicking next dismisses it. The first case fails against the previous code.

File: backend/app/routers/flashcards.py (hunk 6 above)
Comment (arqo123) [reply backend/app/routers/flashcards.py]: The dedup query now selects the row itself (`select(Flashcard) ... order_by(Flashcard.id).limit(1)` → `scalar_one_or_none()`), so there is no separate `db.get` anymore: a concurrent delete simply turns into a miss and the request continues with the normal lookup / insert. Ordering by id makes the winner deterministic (oldest card).

Test `test_create_flashcard_from_word_existing_duplicates_return_oldest` seeds three case / whitespace variants of the same word and asserts the lowest id comes back on repeated calls.

**Comment by arqo123** [comment]
Thanks for the thorough review. All six inline points are addressed in the latest commits; each thread has a reply with the details. In short:

- **Dedup query**: one targeted `lower(trim(word))` lookup per check (before and after the LLM call), fetching the row itself, ordered by id. No full-plan scans, no separate `db.get`, deterministic winner among pre-existing duplicates.
- **Source semantics**: a matching generated/imported card is promoted to `from_text` in place instead of being reported as already saved. `already_saved` now strictly means "already in My Vocabulary".
- **Tooltip lifecycle**: dismissed on transcript mutation, session start and session finalize in `ConversationMode`; lesson page dismisses only when `currentExercise` changes; `useWordSave` ignores responses and timers from a previous selection.
- **Concurrency**: the PR description and the `/from-word` line in `specs/api-endpoints.instructions.md` now describe the dedup as best-effort. A full guarantee needs a normalized per-user/plan uniqueness constraint and a migration; I can open a follow-up PR for that if you want it.
- **Tests**: 3 backend (`source=None` collision before and after the LLM, duplicate determinism), 3 hook tests (B while A pending, dismiss while pending, dismiss-timer cancellation), `ConversationMode` integration (3 cases) and lesson page integration (2 cases). Backend 1026 passed, frontend 498 passed, lint and `tsc` clean.

I only touched the spec line this PR had added, because it described the old dedup behaviour. Changelog and version are untouched as agreed.

**Comment by artcc** [comment]
Thank you for addressing the review so thoroughly and adding the integration tests. I reviewed both follow-up commits through `fc3da33`, and most of the original concerns are now resolved: source promotion, deterministic matching, avoiding full-plan materialization in Python, conversation lifecycle dismissal, and isolation between successive selections. The latest backend and frontend checks are green as well.

Before approval, could you please make three final adjustments?

1. **Use equivalent normalization on both sides of the deduplication comparison.** `_normalize_flashcard_word()` collapses whitespace, whereas `lower(trim(Flashcard.word))` does not. An imported card containing `"take  off"` can therefore fail to match `"take off"` and produce a duplicate without concurrency. I appreciate the single-token restriction in the UI, but the request schema still accepts phrases, and the LLM result can also be a compound term. Please align the normalization and add a regression test for this case.

2. **Invalidate pending save responses when `useWordSave` unmounts.** The cleanup currently clears only an existing dismiss timer. If a save is pending during unmount, its response can still pass the selection-ID check and create a new timer afterward. That timer can subsequently call `removeAllRanges()` on a selection made on the next page. Please invalidate the outstanding selection/request during cleanup and add a test that resolves a pending save after unmount, checking that no new dismiss timer or browser-selection side effect occurs.

3. **Dismiss the lesson tooltip when regeneration actually replaces the question.** Separating dismissal from generic `exercises` updates correctly fixes the answer/hint case. However, regeneration can replace the question while keeping the same `currentExercise` index, leaving the tooltip attached to the previous word/context. Please handle that replacement explicitly, while retaining the tooltip for answer and hint updates that do not change the question. A test selecting a word while regeneration is pending and then resolving the replacement would cover this.

On the third point, I should clarify our previous feedback: grouping regeneration together with answer/hint updates was too broad. Regeneration can change the selected text itself, so this is a refinement of that recommendation, not a request to restore dismissal on every array update.

Two smaller observations, separate from those three adjustments:

- The old ID-query/`db.get()` deletion race is fixed. The new promotion path still has a narrower concurrent-deletion window: deleting the row after it has been read can make the promotion commit or refresh fail. Please consider handling that disappearance; a concurrent delete only becomes a normal lookup miss if it happens before the read.
- The targeted query removes the O(n) transfer and Python materialization, which is a useful improvement. It does not necessarily eliminate scanning the candidate rows in PostgreSQL because the current indexes do not index `lower(trim(word))`. I would not reopen the previous performance request on that basis, but please soften the PR description's “no full-plan scans” claim. A normalized index can remain a follow-up, together with strict concurrent uniqueness.

The changelog, version, specs, and documented test metrics remain the repository owner's responsibility as agreed. Thank you again for the careful follow-up.

**Comment by artcc** [comment]
Thank you again for the work on this PR and for addressing the initial review feedback.

We cannot merge it yet because `develop` has advanced with changes that are already in production, and those changes now conflict with this branch. Could you please bring the latest `develop` into your branch and resolve the conflicts while preserving the production fixes?

The two conflicting files are:

- `frontend/src/components/conversation/ConversationMode.tsx`: both branches modify `finalizeSession()`. Please preserve the current `develop` implementation, including its resource cleanup, microphone/VAD handling, and lifecycle guards, and integrate the tooltip dismissal into that flow.
- `frontend/tests/components/ConversationMode.test.tsx`: both branches independently added this file. Please retain the existing lifecycle coverage from `develop` alongside the new tooltip tests, adapting the mocks to the current microphone acquisition flow rather than replacing either test suite.

The three follow-up adjustments in [the latest review comment](https://github.com/artcc/freelingo/pull/315#issuecomment-5574751190) also remain outstanding on the current head (`fc3da33`): equivalent deduplication normalization, invalidating pending saves on hook unmount, and dismissing the tooltip when regeneration replaces the question.

Once the branch is updated and those adjustments are included, please rerun the relevant checks against the combined code. The current green checks predate the changes now in `develop`, so they do not validate this integration yet. We will review the updated result before merging.

Changelog, version, and specs remain our responsibility as agreed. Thank you for your patience and help getting this integrated without regressing the production behavior.

**Comment by arqo123** [comment]
Merged the latest `develop` into this branch — the conflicts are resolved and the PR is mergeable again.

What needed manual resolution:

- `AGENTS.md` — kept the new bulleted structure from `develop` and re-applied the one-sentence update about the flashcard lookup flow in the learning resources section.
- `ConversationMode.tsx` — kept the hardened `mountedRef` guard and state resets from `develop`, with the tooltip dismissal added inside it.
- `TranscriptBubble.tsx` — kept the new `max-w-[70ch]` typography class alongside the word-selection classes.
- `ConversationMode.test.tsx` — `develop` added a session-lifecycle suite under the same filename. I kept that file untouched and moved the word-tooltip suite to `ConversationModeWordTooltip.test.tsx`, aligning its mocks with the new microphone-before-WebSocket startup order.

Verified locally on the merge commit: `./scripts/format.sh`, `npx tsc --noEmit`, `npm run lint`, `npm run test:run` (513 passing) and `pytest` (1026 passing, 85.58% coverage).

Thanks for the new visual identity work — happy to adjust anything here if you'd prefer a different resolution.
